scipy/stats/_axis_nan_policy.py

```diff
@@ -480,7 +480,8 @@ def hypotest_fun_out(*samples, **kwds):
                     reduced_axes = tuple(range(n_dims))
                 samples = [np.asarray(sample.ravel()) for sample in samples]
             else:
-                samples = _broadcast_arrays(samples, axis=axis)
+                # don't ignore any axes when broadcasting if paired
+                samples = _broadcast_arrays(samples, axis=axis if not paired else None)
```

Comment:
The fix is to ensure that `paired` is considered wherever a `_broadcast` function is used in the decorator.

Reply:
With this change, the decorator broadcasts a sample with length `1` along axis to match the length of the other sample along `axis`. As the failing test highlights, if the length along `axis` was `1` in one sample and different in another sample, an error was raised before; now we get a result. What do we want to do here?

Reply:
@tirthasheshpatel this still needs to be resolved before we can merge this one.

Reply:
Ah, I didn't see this comment. I'd prefer to not broadcast along the `axis` dimension. For some tests, the `axis` dimension has the number of observations which should always match (like `ttest_rel`). Broadcasting doesn't make any sense there.

Reply:
Hmm, I wouldn't say that it doesn't make sense, though. It just shows that having `ttest_rel` and `ttest_1samp` is redundant: `ttest_rel` using broadcasting on the second sample is just `ttest_1samp`.

Reply:
> That said, it won't make much sense for correlation tests.

Yeah, that was the gist of my comment. For users, this might come out as a surprising behaviour for some tests. I don't think any functions in stats allow broadcasting along the axis dimension (not intentionally, at least...)

Reply:
OK. I have a set of changes in mind; lets see if you come up with the same things - where would you perform this check that the arrays have the same length along `axis` if `paired`?

Reply:
@tirthasheshpatel looks like this was waiting on you.